azurerm/internal/services/applicationinsights/resource_arm_application_insights.go

```diff
@@ -167,13 +184,32 @@ func resourceArmApplicationInsightsCreateUpdate(d *schema.ResourceData, meta int
 		return fmt.Errorf("Cannot read AzureRM Application Insights '%s' (Resource Group %s) ID", name, resGroup)
 	}
 
+	billingFeatureRead, err := billingFeatureClient.Get(ctx, resGroup, name)
+	if err != nil {
+		return fmt.Errorf("Error read Application Insights Billing Features %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	applicationInsightsComponentBillingFeatures := insights.ApplicationInsightsComponentBillingFeatures{
+		CurrentBillingFeatures: billingFeatureRead.CurrentBillingFeatures,
+		DataVolumeCap: &insights.ApplicationInsightsComponentDataVolumeCap{
+			Cap:                            dailyCap,
+			StopSendNotificationWhenHitCap: stopSendNotificationWhenHitCap,
+		},
+	}
+
+	_, err = billingFeatureClient.Update(ctx, resGroup, name, applicationInsightsComponentBillingFeatures)
+	if err != nil {
```

Comment:
These lines can be merged

Reply:
Merged if condition and resource update.